src/features/community/useCommunityActions.tsx

```diff
@@ -0,0 +1,204 @@
+import { useIonActionSheet, useIonRouter, useIonToast } from "@ionic/react";
+import { useAppDispatch, useAppSelector } from "../../store";
+import { useBuildGeneralBrowseLink } from "../../helpers/routes";
+import {
+  isAdminSelector,
+  localUserSelector,
+  showNsfw,
+} from "../auth/authSlice";
+import { useContext, useEffect } from "react";
+import { PageContext } from "../auth/PageContext";
+import { checkIsMod } from "../../helpers/lemmy";
+import {
+  addFavorite,
+  blockCommunity,
+  followCommunity,
+  getCommunity,
+  removeFavorite,
+} from "./communitySlice";
+import {
+  allNSFWHidden,
+  buildBlocked,
+  buildProblemSubscribing,
+  buildSuccessSubscribing,
+} from "../../helpers/toastMessages";
+
+export default function useCommunityActions(communityHandle: string) {
+  const [present] = useIonToast();
+  const router = useIonRouter();
+  const dispatch = useAppDispatch();
+  const buildGeneralBrowseLink = useBuildGeneralBrowseLink();
+  const site = useAppSelector((state) => state.auth.site);
+  const isAdmin = useAppSelector(isAdminSelector);
+  const localUser = useAppSelector(localUserSelector);
+  const [presentActionSheet] = useIonActionSheet();
+  const { presentPostEditor } = useContext(PageContext);
+
+  const { presentLoginIfNeeded } = useContext(PageContext);
+
+  const communityByHandle = useAppSelector(
+    (state) => state.community.communityByHandle,
+  );
+
+  useEffect(() => {
+    if (communityByHandle[communityHandle]) return;
+
+    dispatch(getCommunity(communityHandle));
```

Comment:
This is spamming to backend (one call per community, and then calls again multiple times every time you scroll)

This can be tested by searching for community "test" on lemmy.world and then scrolling the page inspecting the network requests.

Additionally, the search response endpoint provides subscribed status per sub. Can we use that so we don't need to make a call per community?

Reply:
I thought it would be cached in `communityByHandle[communityHandle]` :thinking: 
I got the code from view community page

anyways I'll look into these in a day or so!